older git (2.11.0) ls-files does not list cwd which belongs subdataset

While testing git-annex build on jessie (for maximum portability across kernels) using "stack" distribution of ghc and libraries, ran into our safe-guard test failing:
```
======================================================================
FAIL: datalad.distribution.tests.test_dataset.test_is_installed
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 791, in newfunc
    t(*(arg + (uri,)), **kw)
  File "/home/yoh/proj/datalad/datalad/datalad/tests/utils.py", line 615, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad/datalad/distribution/tests/test_dataset.py", line 111, in test_is_installed
    subds.create()
AssertionError: PathKnownToRepositoryError not raised
```

which is due to
```
(git-annex)hopa:~/.tmp/datalad_temp_test_is_installedd9cRgg[master]subm 1
$> /usr/lib/git-annex.linux/git ls-files       
$> /usr/lib/git-annex.linux/git --version
git version 2.11.0
````
as compared to
```shell
$> git ls-files                                
./
$> git --version
git version 2.20.1
```
so it seems we need to check also for minimal git version compatibility :-/  

master version (0.12.0rc2-128-gf9ff64ea) seems to be ok on that test, but now I will run the full battery to see if any other effect

--- a/datalad/support/exceptions.py
+++ b/datalad/support/exceptions.py
@@ -351,3 +351,19 @@ class CrawlerError(Exception):
 
 class PipelineNotSpecifiedError(CrawlerError):
     pass
+
+
+#
+# Warnings
+#
+
+class DataLadWarning(Warning):
+    pass
+
+
+# We have an exception OutdatedExternalDependency, but it is intended for
+# an instance being raised.  `warnings` module requires a class to be provided
+# as a category, so here is a dedicated Warning class
+class OutdatedExternalDependencyWarning(DataLadWarning):
+    """Warning "category" to use to report about outdated"""
+    pass

--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -18,6 +18,7 @@ import shlex
 import time
 import os
 import os.path as op
+import warnings
 from os import linesep
 from os.path import join as opj
 from os.path import exists
@@ -72,6 +73,7 @@ from .exceptions import DeprecatedError
 from .exceptions import FileNotInRepositoryError
 from .exceptions import GitIgnoreError
 from .exceptions import MissingBranchError
+from .exceptions import OutdatedExternalDependencyWarning
 from .exceptions import PathKnownToRepositoryError
 from .network import RI, PathRI
 from .network import is_ssh
@@ -729,7 +731,17 @@ class GitRepo(RepoInterface):
         self._fake_dates_enabled = None
 
     def _create_empty_repo(self, path, **kwargs):
-        if op.lexists(path):
+        # the issue: https://github.com/datalad/datalad/issues/3295
+        # discussion to just issue a warning:
+        #   https://github.com/datalad/datalad/pull/3296
+        if external_versions['cmd:git'] < '2.14.0':
+            warnings.warn(
+                "Your git version (%s) is too old, we will not safe-guard "
+                "against creating a new repository under already known to git "
+                "subdirectory" % external_versions['cmd:git'],
+                OutdatedExternalDependencyWarning
+            )
+        elif op.lexists(path):
             # Verify that we are not trying to initialize a new git repository
             # under a directory some files of which are already tracked by git
             # use case: https://github.com/datalad/datalad/issues/3068

--- a/datalad/support/tests/test_gitrepo.py
+++ b/datalad/support/tests/test_gitrepo.py
@@ -45,6 +45,7 @@ from datalad.tests.utils import assert_cwd_unchanged
 from datalad.tests.utils import local_testrepo_flavors
 from datalad.tests.utils import get_most_obscure_supported_name
 from datalad.tests.utils import SkipTest
+from datalad.tests.utils import skip_if
 from datalad.utils import rmtree
 from datalad.tests.utils_testrepos import BasicAnnexTestRepo
 from datalad.utils import getpwd, chpwd
@@ -147,6 +148,7 @@ def test_GitRepo_init_options(path):
     ok_(cfg.get_value(section="core", option="bare"))
 
 
+@skip_if(external_versions['cmd:git'] < '2.14.0')
 @with_tree(
     tree={
         'subds': {

